### PR TITLE
Chat improvements

### PR DESCRIPTION
**Pull request by roman-drozd-it**
Pull request bringing together several Chat issues.

**What is done/undone:**
- [x] move share file tab functionality into Chat tab
- [x] add saving chat
- [x] add sorting chat
- [x] add combining messages from the same user
- [x] add message formating
- [x] change message appearance
- [x] pin/unpin the Drawer directly from it
- [x] add a goToNewest button with showing an unread messages counter
- [x] highlight new messages
- [x] and other smaller changes
- [x] Add time to every part of Chat combined message
- [x] fix round corners of combined messages
- [x] break spaceless strings longer then chat width
- [x] fix - input loses selecting after clicking a formating button
- [x] [mobile] hide pin/unpin button
- [x] fix combined messages time margins
- [x] fix messages counter, doesn't work on "goToNewest" button when the Chat sorting is "desc" (always shows "0") 
- [x] fix unexpected highlighting all Chat messages when Chat component is refreshing
- [x] unify file item
- [x] fix not working progress bar when downloading a file
- [x] fix unexpected propagation a current message width on the rest of messages when it's wider than the others
- [x] fix [mobile] - saving chat doesn't work [Samsung Internet]
- [x]  ~~add load spinner indicator to file item~~
- [x] fix Chat file progress indicator
- [x] fix dd load spinner indicator to file item~~
- [x] [mobile] fix app crash when erasing a formatted content using backspace key

Any comments/remarks/suggestions are appreciated.

### CODE REVIEW

File: app/src/translations/cs.json [hunk 1]
```diff
@@ -109,6 +109,7 @@
 	"label.leave": "Opustit",
 	"label.chatInput": "Napište zprávu chatu",
 	"label.chat": "Chat",
+	"label.chatNoMessages": null,
```
Comment (jmencak) [inline app/src/translations/cs.json:112]:
```
"label.chatNoMessages": "Žádné zprávy",
```

Comment (test-drozd-it) [reply app/src/translations/cs.json:112]: Added.

Comment (roman-drozd-it) [reply app/src/translations/cs.json]: Added.

**Comment by roman-drozd-it** [comment]
added to todo: [mobile] fix loos of selecting after clicking the formating button

**Comment by roman-drozd-it** [comment]
[mobile] Save chat doesn't work

**Comment by so010** [comment]
> [mobile] Saving chat doesn't work

Perhaps we can disable save chat on mobile.

**Comment by so010** [comment]
[mobile] The pin/unpin is not needed / should be hidden on mobile

**Comment by so010** [comment]
> [mobile] The pin/unpin is not needed / should be hidden on mobile

Same in settings/appearance: "Side drawer over content" - but this is not chat related

**Comment by Alex-YddeR** [comment]
For combined chat the display time is always the first message time. may be we need to display time for each message or disable the combined chat.

**Comment by roman-drozd-it** [comment]
- [x] fix - input loses selecting after clicking a formating button

**Comment by roman-drozd-it** [comment]
- [x]  [mobile] hide pin/unpin button

**Comment by roman-drozd-it** [comment]
- [x] break spaceless strings longer then chat width

**Comment by roman-drozd-it** [comment]
@Alex-YddeR 
> For combined chat the display time is always the first message time. may be we need to display time for each message or disable the combined chat.

- [x] Add time to every part of Chat combined message

**Comment by roman-drozd-it** [comment]
- [x] fix round corners of combined message

**Comment by roman-drozd-it** [comment]
- [ ] [mobile] fix app crash when erasing a formatted content using backspace key

**Comment by roman-drozd-it** [comment]
- [ ] Fix combined messages time spacing

**Comment by roman-drozd-it** [comment]
> * [ ]  [mobile] Saving chat doesn't work

Turned out the problem is on **Samsung Internet** (at least), on **Google Chrome** works corectly

**Comment by roman-drozd-it** [comment]
- [x] Fix Chat combined messages time margins

**Comment by Alex-YddeR** [comment]
I think we can remove (**Guest 93174 (Me) - 11:43 AM**) time from the first time message since it is not so useful to keep bold letter time for every new message and also we are showing time for every message.

**Comment by roman-drozd-it** [comment]
- [x] fix messages counter, doesn't work on "goToNewest" button when the Chat sorting is "desc" (always shows "0")

**Comment by roman-drozd-it** [comment]
- [x] Fix unexpected highlighting all Chat messages when Chat component is refreshing

**Comment by roman-drozd-it** [comment]
> I think we can remove (**Guest 93174 (Me) - 11:43 AM**) time from the first time message since it is not so useful to keep bold letter time for every new message and also we are showing time for every message.

Showing time in very first part of combined message is needed because it always store the first written message.

**Comment by roman-drozd-it** [comment]
- [ ] add load spinner indicator

**Comment by roman-drozd-it** [comment]
[x] unify file item

**Comment by roman-drozd-it** [comment]
- [x] fix not working progress bar when downloading a file

**Comment by roman-drozd-it** [comment]
-[x] fix [mobile] - saving chat doesn't work [Samsung Internet]

**Comment by roman-drozd-it** [comment]
- [x] [mobile] fix app crash when erasing a formatted content using backspace key

**Comment by roman-drozd-it** [comment]
- [x] Fix Chat file progress indicator

**Comment by roman-drozd-it** [comment]
- [x] fix unexpected propagation a current message width on the rest of messages when it's wider than the others

**Pull request merged**